treeherder/perf/auto_perf_sheriffing/outcome_checker.py

```diff
@@ -37,11 +20,8 @@ def __init__(self):
         pass
 
     def check(self, record: BackfillRecord) -> OutcomeStatus:
```

Comment:
```suggestion
    def check(self, record: BackfillRecord) -> OutcomeStatus:
        of_type = record.job_type
        with_unknown_results = 'unknown'
        pushes_in_range = record.get_pushes_in_context_range()

        for push in pushes_in_range:
            # make sure it has at least one successful job of job type
            if push.total_successful_jobs(of_type) == 0:
                # either (at least) one job is in progress or it failed
                if push.total_unfinished_jobs(of_type, with_unknown_results) > 0:
                    return OutcomeStatus.IN_PROGRESS
                else:
                    return OutcomeStatus.FAILED

        return OutcomeStatus.SUCCESSFUL
```

Reply:
What do you think of these cosmetics?

Reply:
Thanks for the suggestions. I added a method to the Push model for getting the total number of jobs of a type with a particular result(`success` or `unknown`).